Fix projectToComPosition

pFernbach opened the pull request:
Update relative to changes in hpp-constraints.

pFernbach committed:
Fix projectToComPosition method

stonneau commented:
Ok so I guess I had not tested it.
Would you mind adding a unit test to test the projection, with HyQ ?
This should be easy using the code here:
https://github.com/humanoid-path-planner/hpp-rbprm/blob/rbprm_v_1/tests/tools-fullbody.hh

stonneau commented:
Also, is the setting of the computation flags necessary at this point ?

pFernbach committed:
remove changes in device computation flags in projection.cc

pFernbach committed:
tools-fullbody : createState correctly compute COM value

pFernbach committed:
[tests] add unit test for projectToComPosition

pFernbach commented:
> Ok so I guess I had not tested it.
> Would you mind adding a unit test to test the projection, with HyQ ?

Done, and I fixed some mistakes in tools-fullbody.hh

> Also, is the setting of the computation flags necessary at this point ?

No, I removed them after some testing.

pFernbach commented:
I still have two issues with this part of the code. 
See https://github.com/humanoid-path-planner/hpp-rbprm/issues/8  for the first one.

For the second : the `ProjectionReport `have a member `status_` which is an enum with the following values : 

* no_contact
* unstable_contact
* stable_contact
* reachable_contact

For some projection (including projectToComPosition) the equilibrium is not tested and this enum always have the value "no_contact". I think we should add a value "unknown".

stonneau commented:
hmm, the projection is successful ?
In this case maybe no_contact should be renamed  failure, and we can add a success value?

pFernbach commented:
Sorry I wasn't clear enough : 
`ProjectionReport` have a member `success` and `status`. But most of the projection method which return a `ProjectionReport` ( except `generate_contact`) never change the field `status` and leave it at it's default value : no_contact

stonneau commented:
ok, so no_contact should just be fail no ?

stonneau commented:
ok sorry I got it after looking at the code. Maybe in this case status should just be contactstatus ?

pFernbach committed:
Remove member com_ from State struct
fix #8

stonneau commented:
I merge the PR since the problem mentioned is just a refactoring issue
